docs/Developers/Debug_Bears.rst

```diff
@@ -122,3 +122,88 @@ Or to debug all bears specified by ``bears`` setting:
     bears = PEP8Bear,MypyBear
     files = <filename>
     debug_bears = True
+
+Debugger Features
+-----------------
+
+Inspect Bear Settings
+^^^^^^^^^^^^^^^^^^^^^
+
+A new command ``settings`` is included in coala's Debugging Interface to
+inspect Bear settings in the debugging environment. It displays all passed
+settings of a Bear and their values so you can quickly inspect them.
+
+Below is an example Bear which detects if a file has more than
+``max_number_of_lines`` lines.
+
+.. code:: python
+
+    from coalib.results.Result import Result
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class TooManyLinesBear(LocalBear):
+
+        def run(self, filename, file, max_number_of_lines: int):
+            """
+            Detects if a file has more than ``max_number_of_lines`` lines.
+
+            :param max_number_of_lines:
+                Maximum number of lines to be allowed for a file.
+            """
+
+            if len(file) > max_number_of_lines:
+                yield Result.from_values(
+                    origin=self,
+                    message=('This file has {} lines, while {} lines'
+                             ' are allowed'
+                             .format(len(file), max_number_of_lines)),
+                    file=filename)
+
+In the above example ``max_number_of_lines`` is a non-optional setting. After
+invoking the debugger on TooManyLinesBear, coala will first ask the value of
+all unpassed non optional setting i.e, ``max_number_of_lines`` and after that
+debugger will get invoke on ``run()`` method of TooManyLinesBear, ``settings``
```

Comment:
*the* debugger will *be* invoke*d* ~~on ``run()`` method of TooManyLinesBear~~. *Then the* ``settings`` command...